### PR TITLE
Remove lingering references to CircleCI which is no longer used.

### PR DESCRIPTION
**Pull request by derekpierre**
**Type of PR:**
- [ ] Bugfix
- [ ] Feature
- [ ] Documentation
- [ ] Other

**Required reviews:** 
- [X] 1
- [ ] 2
- [ ] 3

**What this does:**
Removes some references to Circle CI.

### CODE REVIEW

**Review by KPrasch — COMMENTED** [review]
good cleanup 👍🏻

**Review by theref — DISMISSED** [review]

**Review by cygnusv — DISMISSED** [review]

**Comment by derekpierre** [comment]
Waiting for #3030 to be merged before merging this PR.

**Comment by derekpierre** [comment]
rebased over #3034 .

**Review by KPrasch — APPROVED** [review]

**Pull request merged**
